### ISSUE TITLE
Provisioning machines without an SSH key no longer possible

### ISSUE THREAD

**Issue by echozio** [syself/cluster-api-provider-hetzner#1460]
/kind bug

After #1413 was merged machines that don't use SSH keys fail to provision. One use case in particular for machines without SSH keys is Talos Linux, which is remote controlled over an API and does not use SSH.

I don't think adding an SSH key prevents Talos from working properly, but it seems unnecessary to add one essentially just to make an if statement pass, as earlier versions worked fine with no SSH key.

I'm not aware of the background for this change, and for all I know it's useful to a lot of people, but being able to disable it somehow would be helpful to avoid unnecessary configuration and attack surface in cases where SSH isn't desired.

As for a workaround, downgrading to v1.0.0-beta.41 gets machines provisioned again.

**Environment:**

- cluster-api-provider-hetzner version:  v1.0.0-beta.42
- OS (e.g. from `/etc/os-release`):  Talos Linux

**Comment by janiskemper** [comment]
@batistein you mainly wanted to add this

**Comment by guettli** [comment]
@echozio I see several solutions:

we provide an annotation, if that is set, then the ssh key is optional.

Or we check for the control-plane-provider. If it is not KubeAdm, then the ssh-key is not required.

Or we revert or change.

---

to be honest I am unsure where the ssh-key was made required. For hcloud the ssh-key is not needed for provisioning. For bare-metal ssh access is a hard requirement.
